src/test/java/com/truelayer/java/contract/ContractTests.java

```diff
@@ -0,0 +1,42 @@
+package com.truelayer.java.contract;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+import com.github.tomakehurst.wiremock.common.ConsoleNotifier;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import com.truelayer.java.Environment;
+import com.truelayer.java.TestUtils;
+import com.truelayer.java.TrueLayerClient;
+import java.net.URI;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@Tag("contract")
+public abstract class ContractTests {
```

Comment:
integration tests and contract tests solve different purposes and have different setups. In the existing integration test suite we had very few or zero assertions on requests. In the current integration tests we basically test that the proper endpoints are invoked (no matter how) and that JSON responses are properly deserialized into POJOs. In contract tests instead we want/have to define very strict assertion on the requests, as they have to match whatever documented on our public docs. 

For the above reasons, as of now I ended creating a new test suite specific for contract tests. Once the POC will be completed we will review whether it makes sense to have integration and contract tests implemented in the same test suite. My gut feeling is that the answer is no, right now